### PR TITLE
feat(environment-logs): add deployment history

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

[> Link to the JIRA ticket
](https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-690)

- Add deployment history for Deployment logs with a `versionId` on the routing
Capture d’écran 2023-08-09 à 18 29 05

- Placeholder: service deployed but not on this deployment
Capture d’écran 2023-08-09 à 18 31 16

- Placeholder: service never deployed
Capture d’écran 2023-08-09 à 18 33 09

_Minor_
- Update `Menu` component with new dark color
- Fix `Menu` duplicate className
- Update `dateFullFormat` helper and remove by default millisecond
- Refacto context with serviceId and now use URL params  

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] staging - feat(environment-logs): add deployment history - 2023-08-04T13:03:00Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/11b0ac74-d540-471b-83c9-d37b43681052/services/general)

 Another comment will be posted when deployments are terminated

File: libs/pages/logs/environment/src/lib/feature/deployment-logs-feature/deployment-logs-feature.tsx [hunk 1]
```diff
@@ -57,26 +57,83 @@ export function getServiceStatuesById(services?: DeploymentStageWithServicesStat
...
+
+    const url = `wss://ws.qovery.com/deployment/logs?organization=${organizationId}&cluster=${environment?.cluster_id}&project=${projectId}&environment=${environmentId}&version=${versionId}`
+    const token = await getAccessTokenSilently()
+
+    return new Promise((resolve) => {
+      environment?.cluster_id && resolve(url + `&bearer_token=${token}`)
+    })
+  }, [organizationId, environment?.cluster_id, projectId, environmentId, versionId, getAccessTokenSilently])
+
+  useWebSocket(deploymentLogsUrl, {
```
Comment (ctjhoa) [inline libs/pages/logs/environment/src/lib/feature/deployment-logs-feature/deployment-logs-feature.tsx:97]: We should use common hook based [this hook](https://github.com/Qovery/console/pull/744)

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/shared/ui/src/lib/components/menu/menu.tsx [hunk 2]
```diff
@@ -144,10 +144,10 @@ export function Menu(props: MenuProps) {
         onClose={(e) => handleClick(e)}
         anchorRef={ref}
         align={arrowAlign}
-        className="menu z-20"
-        menuClassName={`rounded-md shadow-[0_0_32px_rgba(0,0,0,0.08)] p-0 ${className} menu__container menu__container--${direction} menu__container--${
+        className="menu z-30"
+        menuClassName={`rounded-md shadow-[0_0_32px_rgba(0,0,0,0.08)] p-0 menu__container menu__container--${direction} menu__container--${
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/shared/ui/src/lib/components/menu/menu.tsx:148]: Ok className is already applied to the parent div

File: libs/shared/utils/src/lib/tools/date.tsx [hunk 3]
```diff
@@ -48,9 +48,9 @@ export function dateYearMonthDayHourMinuteSecond(date: Date, withTime = true, wi
 }
 
 // 15 Sep, 10:23:20:20
-export const dateFullFormat = (date: string, timeZone?: string) => {
+export const dateFullFormat = (date: string, timeZone?: string, customFormat = 'dd MMM, HH:mm:ss') => {
```
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/shared/utils/src/lib/tools/date.tsx:51]: Yep done

File: libs/shared/console-shared/src/lib/layout-logs/layout-logs.tsx [hunk 4]
```diff
@@ -117,7 +117,8 @@ export function LayoutLogs(props: LayoutLogsProps) {
         <div className="absolute z-20 overflow-y-auto left-1 flex items-center w-[calc(100%-8px)] h-11 border-b border-element-light-darker-200 bg-element-light-darker-700">
           {LinkNavigation(
             'Deployment logs',
-            ENVIRONMENT_LOGS_URL(organizationId, projectId, environmentId) + DEPLOYMENT_LOGS_URL(serviceId),
+            ENVIRONMENT_LOGS_URL(organizationId, projectId, environmentId) +
+              DEPLOYMENT_LOGS_VERSION_URL(serviceId, versionId !== ':versionId' ? '' : versionId),
```
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/shared/console-shared/src/lib/layout-logs/layout-logs.tsx]: Fixed, not necessary I forgot to remove this condition, thanks!

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**